src/interface/src/app/services/plan.service.ts

```diff
@@ -351,16 +352,31 @@ export class PlanService {
     return {
       id: scenario.id,
       planId: scenario.plan,
+      projectId: scenario.project,
+      owner: scenario.owner,
       createdTimestamp: this.convertBackendTimestamptoFrontendTimestamp(
         scenario.creation_timestamp
       ),
       priorities: scenario.priorities,
+      projectAreas: this.convertToProjectAreas(scenario.project_areas || []),
       notes: scenario.notes,
-      owner: scenario.owner,
       favorited: scenario.favorited,
     };
   }
 
+  private convertToProjectAreas(scenarioProjectAreas: any[]): ProjectArea[] {
+    if (!scenarioProjectAreas) {
```

Comment:
nit: not sure if this is needed--if scenarioProjectAreas is empty, then the map statement below will map to an empty list

Reply:
The issue is when scenario.project_areas is undefined, which throws an error, so I set up this to catch it just in case. Although I have that call up there "this.convertToProjectAreas(scenario.project_areas || [])".. this is just in case the caller forgets to pass in an empty array.